libtenzir/builtins/operators/yield.cpp

```diff
@@ -0,0 +1,217 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2021 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <tenzir/argument_parser.hpp>
+#include <tenzir/pipeline.hpp>
+#include <tenzir/plugin.hpp>
+
+#include <arrow/api.h>
+
+#include <ranges>
+#include <string_view>
+
+namespace tenzir::plugins::yield {
+
+namespace {
+
+struct unnest {};
+
+using projection = located<variant<std::string, unnest>>;
+
+constexpr auto unnest_idx = std::numeric_limits<offset::value_type>::max();
+
+class yield_operator final
+  : public schematic_operator<yield_operator,
+                              std::optional<std::pair<offset, type>>> {
+public:
+  yield_operator() = default;
+
+  explicit yield_operator(parser_interface& p) {
+    auto parser = argument_parser{"yield", "https://docs.tenzir.com/next/"
+                                           "operators/transformations/yield"};
+    auto extractor = located<std::string>{};
+    // TODO: This assumes that the given extractor can be parsed as a shell-like
+    // argument, e.g., spaces must be quoted.
+    parser.add(extractor, "<extractor>");
```

Comment:
I think we need to discuss whether we want `.'foo bar'.baz` or `.["foo bar"].baz`. Both cases are also a strong argument for requiring fields to start with a `.` to make parsing easier.

I think we can postpone this and merge the operator regardless, though.

Reply:
Agreed. Let's make this part of the language revamp.